devito/finite_differences/tools.py

```diff
@@ -159,51 +156,46 @@ def symbolic_weights(function, deriv_order, indices, dim):
             for j in range(0, len(indices))]
 
 
-def generate_indices(func, dim, diff, order, stagger=None, side=None):
-
+def generate_indices(func, dim, order, side=None, x0={}):
     # If staggered finited difference
-    if func.is_Staggered:
-        if stagger == left:
-            off = -.5
-        elif stagger == right:
-            off = .5
-        else:
-            off = 0
-        ind = list(set([(dim + int(i+.5+off) * dim.spacing)
-                        for i in range(-order//2, order//2)]))
-        x0 = (dim + off*diff)
-        if order < 2:
-            ind = [dim + diff, dim] if stagger == right else [dim - diff, dim]
-
-        return ind, x0
-
-    # Check if called from first_derivative()
-    if bool(side):
-        if side == right:
-            ind = [(dim+i*diff) for i in range(-int(order/2)+1-(order % 2),
-                                               int((order+1)/2)+2-(order % 2))]
-        elif side == left:
-            ind = [(dim-i*diff) for i in range(-int(order/2)+1-(order % 2),
-                                               int((order+1)/2)+2-(order % 2))]
-        else:
-            ind = [(dim+i*diff) for i in range(-int(order/2),
-                                               int((order+1)/2)+1)]
-        x0 = None
+    if func.is_Staggered and not dim.is_Time:
+        x0, ind = indices_staggered(func, dim, order, side=side, x0=x0)
```

Comment:
Im not sure I appreciate the need for this extra function

Reply:
Makes it a lot easier to work on/maintain so I prefer it

Reply:
ah no sorry.... the name was deceptive...this is basically just `generate_indices` demanding the *generation* of the staggered indices to another function ? I think I get it now...

Reply:
yes

Reply:
nitpicking: maybe call them `generate_indices_staggered` and `generate_indices_cartesian`